test/views/pages/profile/delete_post_profile_test.dart

```diff
@@ -0,0 +1,111 @@
+import "package:cloud_firestore/cloud_firestore.dart";
+import "package:fake_cloud_firestore/fake_cloud_firestore.dart";
+import "package:flutter_test/flutter_test.dart";
+import "package:hooks_riverpod/hooks_riverpod.dart";
+import "package:proxima/models/database/post/post_firestore.dart";
+import "package:proxima/models/database/user/user_firestore.dart";
+import "package:proxima/views/pages/profile/info_cards/profile_info_card.dart";
+import "package:proxima/views/pages/profile/info_cards/profile_info_pop_up.dart";
+import "package:proxima/views/pages/profile/profile_data/profile_user_posts.dart";
+
+import "../../../mocks/data/firestore_post.dart";
+import "../../../mocks/data/firestore_user.dart";
+import "../../../mocks/data/geopoint.dart";
+import "../../../mocks/providers/provider_profile_page.dart";
+import "../../../mocks/services/setup_firebase_mocks.dart";
+
+void main() {
+  const delayNeededForAsyncFunctionExecution = Duration(seconds: 1);
```

Comment:
I was wondering if the `delayNeededForAsyncFunctionExecution` is truly necessary in the tests.
Because I tried to remove it and all the tests passed successfully thus I don't really know if we need them ?

Reply:
Yes, so my philosophy of whether to add this or not is basically when an async function is called by the viewmodel, I add the delay just to make sure we don't break future behavior that might arise.

We can remove them if you prefer to only have them where they are absolutely necessary, but I don't really see an argument why we should ?
Let me know what you prefer.

Reply:
I actually wonder what the idea behind this is too. Is the point to reduce the total timeout of the animation?Shouldn't you then do this instead.

```
await tester.pumpAndSettle(const Duration(milliseconds: 100), EnginePhase.sendSemanticsUpdate, delayNeededForAsyncFunctionExecution);
```

From what I understand the following instead increases the duration of the pump, which allows for less reactive widgets to pass.
```
await tester.pumpAndSettle(delayNeededForAsyncFunctionExecution);
```

Reply:
In a similar way (and for the same reason), we always call `tester.pumpAndSettle` even though `test.pump` would work for many cases.

Reply:
Oh ok I see, thanks for your answer.
My worry is that I feel like it will make the tests take a longer time to pass if I understand correctly ?

Reply:
I multiplied by 10 the number of seconds in `delayNeededForAsyncFunctionExecution` and it passed all the tests with 2 seconds less than with 1 second value on my machine (:rofl:).

Don't take this as a proper performance benchmark, of course, but I don't think it has a meaningful impact on the performances of the test suite.

Reply:
I also tried out of curiosity with `100 seconds` and I did not observe any difference in the timing measurements of the test suite.

Reply:
Ok, thank you for testing.
Then it's all good to me :+1:

Reply:
See issue #201 for further discussion on the topic.